connectors/src/connectors/microsoft/temporal/activities.ts

```diff
@@ -26,21 +46,258 @@ export async function fullSyncActivity({
   const client = await getClient(connector.connectionId);
 
   const teamResources = resources.filter((resource) =>
-    ["channel"].includes(resource.nodeType)
+    ["folder"].includes(resource.nodeType)
   );
 
-  const channel = teamResources[0];
+  const folder = teamResources[0];
 
-  if (!channel) {
+  if (!folder) {
     throw new Error(`No channel found for connector ${connectorId}`);
   }
 
   // get a message
-  const message = getMessages(client, channel.itemApiPath);
+  const filesAndFolders = await getFilesAndFolders(
+    client,
+    microsoftInternalIdFromNodeData(folder)
+  );
+
+  logger.info({ filesAndFolders, folder }, "Files and folders for folder");
+  const file = filesAndFolders.filter((item) => item.file)[0];
+  if (!file) {
+    throw new Error(`No file found for folder ${folder.id}`);
+  }
+
+  const startSyncTs = Date.now();
+
+  await syncOneFile(client, {
+    connectorId,
+    dataSourceConfig,
+    file,
+    parent: folder,
+    startSyncTs,
+  });
+
+  if (!file?.id) {
+    throw new Error(`No file or no id`);
+  }
+}
 
-  logger.info(
-    `To implement: full sync for connector ${connectorId} with config ${JSON.stringify(
-      dataSourceConfig
-    )}\nExample message: ${JSON.stringify(message)}`
+export async function syncOneFile(
+  client: Client,
+  {
+    connectorId,
+    dataSourceConfig,
+    file,
+    parent,
+    startSyncTs,
+    isBatchSync = false,
+  }: {
+    connectorId: ModelId;
+    dataSourceConfig: DataSourceConfig;
+    file: microsoftgraph.DriveItem;
+    parent: MicrosoftRootResource;
+    startSyncTs: number;
+    isBatchSync?: boolean;
+  }
+) {
+  const localLogger = logger.child({
+    provider: "microsoft",
+    connectorId: parent.connectorId,
+    internalId: file.id,
+    name: file.name,
+  });
+
+  if (!file.file) {
+    throw new Error(`Item is not a file: ${JSON.stringify(file)}`);
+  }
+
+  const itemApiPath = getDriveItemApiPath(
+    file,
+    microsoftInternalIdFromNodeData(parent)
   );
+
+  const documentId = microsoftInternalIdFromNodeData({
+    itemApiPath,
+    nodeType: "file",
+  });
+
+  const fileResource =
+    await MicrosoftNodeResource.fetchByInternalId(documentId);
+
+  // Early return if lastSeenTs is greater than workflow start.
+  // This allows avoiding resyncing already-synced documents in case of activity failure
+  if (
+    fileResource?.lastSeenTs &&
+    fileResource.lastSeenTs > new Date(startSyncTs)
+  ) {
+    return true;
+  }
+
+  if (fileResource?.skipReason) {
+    localLogger.info(
+      { skipReason: fileResource.skipReason },
+      "Skipping file sync"
+    );
+    return false;
+  }
+
+  const url =
```

Comment:
you can also get the file content by using the endpoint `drives/{{drive-id}}/items/{{file-id}}/content` - i think it's safer and easier in terms of authentication (it uses the authenticated client)

Reply:
As discussed IRL: had tried, didn't work -- the client returns a 302 to the download URL (which is probably the same as the one already provided here